dsl_parser/tests/test_inputs.py

```diff
@@ -1152,80 +1154,121 @@ def test_input_validate_type_raises(self):
+        self._test_input_validate_type_raises_with_deployment('list', '123')
+
+    def test_input_validate_dict_successful(self):
+        self._test_input_validate_type_successful('dict', {'k': 'v'},
+                                                  self.assertDictEqual)
+
+    def test_input_validate_dict_raises(self):
+        self._test_input_validate_type_raises('dict', [])
+
+    def test_input_validate_dict_raises_with_deployment(self):
+        self._test_input_validate_type_raises_with_deployment('dict', 'x')
+
+    def test_input_validate_integer_successful(self):
+        self._test_input_validate_type_successful('integer', 123,
+                                                  self.assertEqual)
+        self._test_input_validate_type_successful('integer', long(),
+                                                  self.assertEqual)
+
+    def test_input_validate_integer_raises(self):
+        self._test_input_validate_type_raises('integer', [])
+        self._test_input_validate_type_raises('integer', True)
+
+    def test_input_validate_integer_raises_with_deployment(self):
+        self._test_input_validate_type_raises_with_deployment('integer', 'x')
+        self._test_input_validate_type_raises_with_deployment('integer', True)
+
+    def test_input_validate_float_successful(self):
+        self._test_input_validate_type_successful('float', 123,
+                                                  self.assertEqual)
+        self._test_input_validate_type_successful('float', 123.1,
+                                                  self.assertEqual)
+        self._test_input_validate_type_successful('float', long(),
+                                                  self.assertEqual)
+
+    def test_input_validate_float_raises(self):
+        self._test_input_validate_type_raises('float', [])
+        self._test_input_validate_type_raises('float', True)
+
+    def test_input_validate_float_raises_with_deployment(self):
+        self._test_input_validate_type_raises_with_deployment('float', 'x')
+        self._test_input_validate_type_raises_with_deployment('float', True)
+
+    def test_input_validate_boolean_successful(self):
+        self._test_input_validate_type_successful('boolean', False,
+                                                  self.assertEqual)
+
+    def test_input_validate_boolean_raises(self):
+        self._test_input_validate_type_raises('boolean', [])
+
+    def test_input_validate_boolean_raises_with_deployment(self):
+        self._test_input_validate_type_raises_with_deployment('boolean', 'x')
+
+    def test_input_validate_string_successful(self):
+        self._test_input_validate_type_successful('string', 'some_string',
+                                                  self.assertEqual)
+        # Using type: string, allows us to pass any kind of type (due to
+        # backward compatibility)
+        self._test_input_validate_type_successful('string', list(),
```

Comment:
why does "list()" works as a type name?

Reply:
It's the the type name, it's the value, i.e. an empty list.